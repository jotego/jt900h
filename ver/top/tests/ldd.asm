    ; LDD
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xhl,data+3*2
    ld xde,copy+3*2
    ld bc,4

    lddw (xde-),(xhl-)
    lddw (xde-),(xhl-)
    lddw (xde-),(xhl-)
    lddw (xde-),(xhl-)

    cp bc,0
    jp ne,bad_end

    ; check the copy
    ld bc,4
    ld xhl,data+3*2
    ld xde,copy+3*2
loop:
    ld wa,(xhl)
    sub xhl,2
    cpd wa,(xde-)
    jp ne,bad_end
    cp bc,0
    jp ne,loop

test_end:
    ; ld (0xffff),0xff
end_loop:
    ldf 0
    ld hl,0xbabe
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
    db 0
data:
    dw 0x0011,0x2233,0x4455,0x6677,0x8899,0x0000,0x0000,0x0000
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
copy:
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
    end