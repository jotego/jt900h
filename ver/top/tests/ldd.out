00000011 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000073 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000000
SR = 42f8
